File: leetsync/1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int n = s.length();
        int result = Integer.MIN_VALUE;
        int zeros = 0;
        int ones = 0;
        for (int i = 0; i < n; i++) {
            char ch = s.charAt(i);
            if (ch == '1') {
                ones++;
            }
        }
        for (int i = 0; i < n - 1; i++) {
            char ch = s.charAt(i);
            if (ch == '0') {
                zeros++;
            } else {
                ones--;
            }
            result = Math.max(result, zeros + ones);
        }
        return result;
    }
}

File: leetsync/1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.go
func maxScore(s string) int {
	n := len(s)
	zeros, ones := 0, 0
	for _, v := range s {
		if v == '1' {
			ones++
		}
	}
	result := 0
	for i := 0; i < n-1; i++ {
		if s[i] == '0' {
			zeros++
		} else {
			ones--
		}
		result = max(result, zeros+ones)
	}
	return result
}